core/src/test/scala/com/mesosphere/usi/core/integration/SchedulerFuzzingTest.scala

```diff
@@ -0,0 +1,155 @@
+import com.mesosphere.usi.core.SchedulerFactory
+import com.mesosphere.usi.core.conf.SchedulerSettings
+import com.mesosphere.usi.core.helpers.SchedulerStreamTestHelpers.commandInputSource
+import com.mesosphere.usi.core.models.{PodId, PodSpecUpdatedEvent, PodStatus, PodStatusUpdatedEvent, StateEvent, TaskId}
+import com.mesosphere.usi.core.models.commands.{ExpungePod, KillPod, LaunchPod, SchedulerCommand}
+import com.mesosphere.usi.core.models.resources.ScalarRequirement
+import com.mesosphere.usi.core.models.template.SimpleRunTemplateFactory
+import com.mesosphere.usi.core.util.DurationConverters
+import com.mesosphere.utils.AkkaUnitTest
+import com.mesosphere.utils.mesos.MesosClusterTest
+import com.mesosphere.utils.metrics.DummyMetrics
+import com.mesosphere.utils.persistence.InMemoryPodRecordRepository
+import org.apache.mesos.v1.Protos
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalatest.Inside
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+import scala.collection.concurrent
+import scala.collection.mutable
+import scala.concurrent.duration._
+
+class SchedulerFuzzingTest extends AkkaUnitTest with MesosClusterTest with Inside with ScalaCheckPropertyChecks {
+  override def materializerSettings = super.materializerSettings.withDebugLogging(true)
+  override implicit lazy val mat = ActorMaterializer()
+
+  lazy val mesosClientSettings = MesosClientSettings.load().withMasters(Seq(mesosFacade.url))
+  val frameworkInfo = Protos.FrameworkInfo
+    .newBuilder()
+    .setUser("test")
+    .setName("SimpleHelloWorldExample")
+    .addRoles("test")
+    .addCapabilities(
+      Protos.FrameworkInfo.Capability.newBuilder().setType(Protos.FrameworkInfo.Capability.Type.MULTI_ROLE))
+    .build()
+
+  "random commands" in {
+    // Expected pod states
+    val expectedState = mutable.Map[PodId, Protos.TaskState]()
+
+    /**
+      * Updated the expected state. It is called in the beginning of forAll.
+      * @param cmd The next command.
+      */
+    def updateExpectedState(cmd: SchedulerCommand): Unit = cmd match {
+      case launchPod: LaunchPod => expectedState.update(launchPod.podId, Protos.TaskState.TASK_RUNNING)
+      case killPod: KillPod =>
+        // Killed pods are automatically expunged once they become terminal.
+        expectedState.remove(killPod.podId)
+      case expungePod: ExpungePod => expectedState.remove(expungePod.podId)
+    }
+
+    // Observed pod states as reported by USI
+    var observedState = concurrent.TrieMap.empty[PodId, Protos.TaskState]
+
+    /**
+      * Process [[StateEvent]]s from USI to replicated the internal USI state.
+      * @param update The update from USI. We only case for [[PodSpecUpdatedEvent]] and [[PodStatusUpdatedEvent]].
+      */
+    def updateObservedState(update: StateEvent): Unit = update match {
+      case PodSpecUpdatedEvent(podId, None) =>
+        logger.debug(s"Removing pod. podId=${podId.value}")
+        observedState.remove(podId)
```

Comment:
This is wrong. We need to track pod statuses as the records stick around but not the specs.